src/site/layout/css.rs

```diff
@@ -28,27 +35,99 @@ fn concat_minify(css_files: &[String]) -> Result<String> {
+    // by falling back to fetching said version from GitHub.
+    if cfg!(debug_assertions) && release_tag == ORANDA_CSS_TAG {
+        // If we're running in a local environment, we fetch a Tailwind binary and compile the CSS
+        // on the spot. This is useful if we're working on oranda-css locally.
+        build_css(dist_dir)
+    } else {
+        // If we're running in a released binary, we fetch the latest tag (or whichever tag is
+        // specified in the configuration).
+        fetch_css(dist_dir, release_tag)
+    }
+}
+
+pub fn build_css(dist_dir: &str) -> Result<()> {
+    // Fetch our cache dir
+    let project_dir = ProjectDirs::from("dev", "axo", "oranda")
+        .expect("Unable to create cache dir for downloading Tailwind!");
+    let cache_dir = project_dir.cache_dir();
+    // Figure out our target "double" (tailwind has weird naming around this)
+    let double = match (env::consts::OS, env::consts::ARCH) {
+        ("linux", "x86_64") => "linux-x64",
+        ("linux", "aarch64") => "linux-arm64",
+        ("linux", "arm") => "linux-armv7",
+        ("macos", "x86_64") => "macos-x64",
+        ("macos", "aarch64") => "macos-arm64",
+        ("windows", "x86_64") => "windows-x64.exe",
+        ("windows", "aarch64") => "windows-arm64.exe",
+        _ => "linux-x64",
+    };
+    let mut binary_path = Utf8PathBuf::from(cache_dir.display().to_string());
+    LocalAsset::create_dir_all(&binary_path)?;
+    binary_path.push(format!("tailwindcss-{double}"));
+    if !binary_path.exists() {
+        // Fetch the binary from GitHub if it doesn't exist
+        tracing::info!("Fetching Tailwind binary from GitHub release...");
+        let url = format!(
+            "https://github.com/tailwindlabs/tailwindcss/releases/latest/download/tailwindcss-{double}"
+        );
+        let handle = tokio::runtime::Handle::current();
+        let response = handle.block_on(reqwest::get(url))?;
+        let bytes = handle.block_on(response.bytes())?;
+        let file = LocalAsset::new(&binary_path, Vec::from(bytes))?;
+        file.write(
+            binary_path
+                .parent()
+                .expect("Tailwind binary path has no parent!?"),
+        )?;
+
+        // On non-Windows platforms, we need to mark the file as executable
+        if !double.starts_with("windows") {
+            Command::new("chmod")
+                .args(["+x", binary_path.as_str()])
+                .output()?;
+        }
+    }
+
+    tracing::info!("Building oranda CSS using Tailwind...");
+    let css_src_path = Utf8PathBuf::from(CSS_SRC_PATH);
+    let output = Command::new(binary_path)
+        .args([
+            "-c",
+            "oranda-css/tailwind.config.js",
+            "-i",
+            css_src_path.as_str(),
+            "-o",
+            &format!("{dist_dir}/oranda.css"),
+            "--minify",
+        ])
+        .output()?;
```

Comment:
Might also want to check `output.status.success()`. Command returning an Err is generally an OS error like "binary_path wasn't executable at all".